road/stats.py

```diff
@@ -0,0 +1,306 @@
+import os
+import pandas as pd
+import numpy as np
+import json
+import yaml
+
+
+def prepare_time_period_dfs(est_df, obs_df, times, combined_df_cols):
+    error = {col: est_df[col] - obs_df[col] for col in times}
+    error_df = pd.DataFrame(error)
+    error_squared_df = error_df.pow(2)
+    error_percent_df = error_df.divide(obs_df[times]).fillna(0)
+    error_percent_df = error_percent_df.apply(lambda x: x * 100, axis=1)
+
+    combined_df = pd.concat(
+        [est_df, obs_df[times], error_df, error_squared_df, error_percent_df], axis=1)
+    combined_df = combined_df.drop_duplicates(subset=['A', 'B'], keep='first')
+
+    calculation_cols = [
+        'Estimated Volume',
+        'Observed Volume',
+        'Errors',
+        'Squared Errors',
+        'Percent Errors']
+
+    time_period_dfs = {}
+
+    for period in times:
+        matching_column_indices = [
+            i for i, col_name in enumerate(
+                combined_df.columns) if col_name == period]
+        select_time_period_df = combined_df.iloc[:, matching_column_indices]
+
+        specific_columns_combined_df = combined_df[combined_df_cols]
+
+        select_time_period_loc_df = pd.concat(
+            [specific_columns_combined_df, select_time_period_df], axis=1)
+
+        new_column_names = combined_df_cols + \
+            calculation_cols[:len(select_time_period_loc_df.columns) - len(combined_df_cols)]
+        select_time_period_loc_df.columns = new_column_names
+
+        time_period_dfs[period] = select_time_period_loc_df
+
+    return time_period_dfs
+
+
+def classify_observation_volume(volume):
+    if volume < 10000:
+        return '<10k'
+    elif 10000 <= volume < 20000:
+        return '10-20k'
+    elif 20000 <= volume < 50000:
+        return '20-50k'
+    else:
+        return '>=50k'
+
+
+# Define function to reset index and rename columns
+def reset_index_and_rename(df, group_var):
+    df.index.name = group_var
+    df = df.reset_index()
+    return df
+
+
+# Define function to reorder DataFrame based on group_var
+def reorder_dataframe(df, group_var):
+    area_type_order = ['Core/CBD', 'UrbBiz', 'Urb', 'Sub', 'All Locations']
+    facility_type_order = ['Fwy/Ramp', 'Art', 'Col', 'Loc', 'All Locations']
+    volume_classification_order = [
+        '<10k',
+        '10-20k',
+        '20-50k',
+        '>=50k',
+        'All Locations']
+
+    if group_var == 'AT Group':
+        order = area_type_order
+    elif group_var == 'FT Group':
+        order = facility_type_order
+    elif group_var == 'Observed Volume':
+        order = volume_classification_order
+    else:
+        return df  # No reordering needed for other group_var values

```

Comment:
this return here kinda functions as a goto statement, reducing readability. Is there a way to do this where there's only a single return at the end? Or at least something that puts the 2 returns at the same indentation/nesting level?